### PR TITLE
Refactoring privacy consent pr2155

### LINKED ISSUE #2155: Update Privacy Consent Info Form
Update PrivacyConsentInfoForm.js to use Formik, Yup, and Bootstrap as documented in issue #2136. These changes have been made to HowDidItStartForm.js, you can them here - #2138.  

This form is a little too complicated for a statement, checkbox, and a conditional error message. It needs to be simplified. Checkbox component has already been created. We could create a link component (which are just buttons in Bootstrap) - documentation [here](https://react-bootstrap.github.io/components/buttons/).

### PR DESCRIPTION
**Pull request by dengcn**
Fixes #2155 

Refactoring Privacy Consent Form

# Description
This is to refactor Privacy Consent Form using new form framework
> Please include a summary of the change.
Privacy consent form related
# Any new packages installed?

> Give details

# Required followup work

> Is there anything related to this that still needs to be done (ex: documentation changes).

# Checklist:

- [ ] I have updated the azurescript.sh with any **new environment variables** and added them to the appsettings
- [ X] I have looked at my code on GitHub and it all looks good (ex: no random commented out code or console.logs)
- [ ] I have added and needed tests for my changes (in particular for new screens)
- [ ] I have added a comment to any confusing code
- [ ] I have added documentation to any modified front-end code. (Or added missing documentation)

### CODE REVIEW

**Comment by dengcn** [comment]
The href to external link and validation are pending on delivery of another PR2269, will update these 2 once PR2269 merged to master

File: f2/src/forms/ContactInfoForm.js [hunk 1]
```diff
@@ -14,27 +14,33 @@ import { Field } from '../components/Field'
...
+
+  const phone =
+    'phone' in values &&
+    containsData(values.phone) &&
+    new RegExp(phoneRegex).test(values.phone)
+  const email =
+    'email' in values &&
+    containsData(values.email) &&
+    addrs(values.email) !== null
+
```
Comment (kevinyang06) [inline f2/src/forms/ContactInfoForm.js:37]: why do contact info form is modified under this PR?

**Comment by ying-chen-cds** [comment]
There seems no validation, even if you do not check the checkbox, you can continue

**Comment by ying-chen-cds** [comment]
The sentence is not complete
privacyConsent

**Comment by dengcn** [comment]
Added followings:
- Embedded link "href" in label, solved the incomplete sentence
- Added validation if checkout not checked, 
- Error message will pop up if user does not check the box and try to move to next page

Note: ignore ContactForm change, this is irrelevant, might be brought in by git

File: f2/src/forms/PrivacyConsentInfoForm.js [hunk 2]
```diff
@@ -1,109 +1,64 @@
...
+            </Row>
+            <Row className="form-section">
+              <Field
+                name="consentOptions"
+                label={
+                  <Trans id="privacyConsentInfoForm.yes.withExternalLink">
+                    <A
+                      color="#0000ff"
+                      target="_blank"
+                      href={'/privacystatement?lang=en'}
```
Comment (justinr86) [inline f2/src/forms/PrivacyConsentInfoForm.js:40]: This link defaults to the English privacy statement. The link should use the language option the user selected.

File: f2/src/forms/PrivacyConsentInfoForm.js [hunk 3]
```diff
@@ -1,109 +1,64 @@
...
-                      </A>
-                      <Trans id="privacyConsentInfoForm.period" />
-                    </CheckboxAdapter>
-                  </React.Fragment>
-                )
-              })}
-            </FormArrayControl>
-            <NextAndCancelButtons
-              next={
-                fyiForm ? (
```
Comment (justinr86) [inline f2/src/forms/PrivacyConsentInfoForm.js:92]: The next page label is hard-coded. The next page is different if the user selects the FYI flow.

File: f2/src/forms/PrivacyConsentInfoForm.js [hunk 4]
```diff
@@ -1,109 +1,64 @@
...
+      initialValues={whetherConsent}
+      onSubmit={(values) => {
+        props.onSubmit(values)
+      }}
+      validationSchema={PrivacyConsentInfoFormSchema()}
+    >
+      {({ handleSubmit, handleChange, handleBlur }) => (
+        <Form onSubmit={handleSubmit}>
+          <Container>
+            <Row className="form-question">
```
Comment (justinr86) [inline f2/src/forms/PrivacyConsentInfoForm.js:29]: We should keep the error messages the same. Please use the ErrorSummary component, see WhenDidItHappen.js

**Comment by dengcn** [comment]
In the other PR , such error component is also used for error handling like this one "https://github.com/cds-snc/report-a-cybercrime/pull/2243/files" , just wonder.

**Comment by justinr86** [comment]
> In the other PR , such error component is also used for error handling like this one "https://github.com/cds-snc/report-a-cybercrime/pull/2243/files" , just wonder.

This form will need to be updated as well (actually original form didn't even have validation so this should probably just be removed.). The new components are being refined as we develop. After discussing with Norman and John it was decided to use an ErrorSummary component. See WhenDidItHappenForm.js and WhatWasAffected.js for example on adding the ErrorSummary object.

File: f2/src/forms/PrivacyConsentInfoForm.js [hunk 5]
```diff
@@ -37,7 +53,7 @@ export const PrivacyConsentInfoForm = (props) => {
                     <A
                       color="#0000ff"
                       target="_blank"
-                      href={'/privacystatement?lang=en'}
+                      href={'/privacystatement?lang=' + i18n.locale} //{'/privacystatement?lang=en'}
```
Comment (justinr86) [inline f2/src/forms/PrivacyConsentInfoForm.js:64]: If this code works we should remove the commented out section.

**Comment by justinr86** [comment]
ErrorSummary needs some work, please see WhenDidItHappenForm.js and WhatWasAffect.js for example.

errorSummary

File: f2/src/forms/PrivacyConsentInfoForm.js [hunk 6]
```diff
@@ -1,109 +1,91 @@
...
+                value="privacyConsentInfoForm.yes"
+                onChange={handleChange}
+                onBlur={handleBlur}
+                type="checkbox"
+                id="consentOptions"
+              ></Field>
+            </Row>
+            <Row>
+              <NextCancelButtons
+                submit={<Trans id="privacyConsentInfoForm.nextButton" />}
```
Comment (justinr86) [inline f2/src/forms/PrivacyConsentInfoForm.js:81]: The next page differs in the FYI flow. Need to change the label depending on report type.

File: f2/src/forms/PrivacyConsentInfoForm.js [hunk 7]
```diff
@@ -1,109 +1,91 @@
...
+            </Row>
+            <Row className="form-section">
+              <Field
+                name="consentOptions"
+                label={
+                  <Trans id="privacyConsentInfoForm.yes.withExternalLink">
+                    <A
+                      color="#0000ff"
+                      target="_blank"
+                      href={'/privacystatement?lang=' + i18n.locale} //{'/privacystatement?lang=en'}
```
Comment (justinr86) [inline f2/src/forms/PrivacyConsentInfoForm.js:67]: Should remove the commented out code.

File: f2/src/forms/PrivacyConsentInfoForm.js [hunk 8]
```diff
@@ -1,109 +1,91 @@
...
+                  submissions={submitCount}
+                  title={
+                    <Trans id="privacyConsentInfoForm.hasValidationErrors" />
+                  }
+                />
+              )}
+            </Row>
+            <Row className="form-question">
+              <ErrorMessage
+                name="privacyConsentInfoForm.warning"
```
Comment (justinr86) [inline f2/src/forms/PrivacyConsentInfoForm.js:55]: This ErrorMessage is not used, it should be removed.

**Review by justinr86 — APPROVED** [review]

**Pull request merged**